### PR TITLE
[i2c target] Add I2C Wakeup Tests 

### PR DESCRIPTION
**Pull request by cfrantz**
1. Put the chip into normal sleep and then issue an I2C transaction to
   wake up the chip.
2. Put the chip into deep sleep and then issue an I2C transaction to
   wake up the chip.

This PR builds on #18453.  You need only review the last commit.

### CODE REVIEW

File: sw/device/tests/i2c_target_test.c [hunk 1]
```diff
@@ -49,14 +55,94 @@ status_t configure_device_address(ujson_t *uj, dif_i2c_t *i2c) {
...
+  while (true) {
+    if (txn->length % 64 == 0 && delay_micros) {
+      busy_spin_micros(delay_micros);
+    }
+    TRY(wait_for_acq_fifo(i2c, 1, &deadline));
+    TRY(dif_i2c_acquire_byte(i2c, &byte, &signal));
+    if (signal != kDifI2cSignalNone) {
+      break;
+    }
+    txn->data[txn->length++] = byte;
```
Comment (engdoreis) [inline sw/device/tests/i2c_target_test.c:136]: Should we check the array boundaries here?

File: sw/device/tests/i2c_target_test.c [hunk 2]
```diff
@@ -124,11 +167,72 @@ static status_t write_transaction(ujson_t *uj, dif_i2c_t *i2c,
...
+  };
+  TRY(dif_pinmux_wakeup_detector_enable(&pinmux, 0, wakeup_cfg));
+  dif_pwrmgr_domain_config_t pwrmgr_domain_cfg = 0;
+  if (normal) {
+    // Normal sleep wakes up from an interrupt, so enable the relevant sources.
+    // Enable all the AON interrupts used in this test.
+    rv_plic_testutils_irq_range_enable(&plic, kTopEarlgreyPlicTargetIbex0,
+                                       kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+                                       kTopEarlgreyPlicIrqIdPwrmgrAonWakeup);
+    // Enable pwrmgr interrupt
```
Comment (engdoreis) [inline sw/device/tests/i2c_target_test.c:185]: Nit: For consistency 
```suggestion
    // Enable pwrmgr interrupt.
```

File: sw/device/tests/i2c_target_test.c [hunk 3]
```diff
@@ -124,11 +167,72 @@ static status_t write_transaction(ujson_t *uj, dif_i2c_t *i2c,
...
+    TRY(dif_pwrmgr_irq_set_enabled(&pwrmgr, 0, kDifToggleEnabled));
+
+    // Configure the power domains for normal sleep;
```
Comment (engdoreis) [inline sw/device/tests/i2c_target_test.c:188]: Nit. 
```suggestion
    // Configure the power domains for normal sleep.
```

**Review by engdoreis — APPROVED** [review]
LGTM.

**Review by engdoreis — APPROVED** [review]
LGTM.

File: sw/device/tests/i2c_target_test.c (hunk 1 above)
Comment (cfrantz) [reply sw/device/tests/i2c_target_test.c:136]: No need: `length` is a uint8_t and can never overflow `data`.

File: sw/device/tests/i2c_target_test.c (hunk 2 above)
Comment (cfrantz) [reply sw/device/tests/i2c_target_test.c]: Fixed.

File: sw/device/tests/i2c_target_test.c (hunk 3 above)
Comment (cfrantz) [reply sw/device/tests/i2c_target_test.c]: Fixed.

**Pull request merged**